### DISCUSSION TITLE
Feature suggestions

### DISCUSSION THREAD

**Discussion by mihaitodor** [davep/oshit#13] [Ideas]
I have some more ideas (no need to implement any of them, they're just suggestions):

- Search through items from the current tab vim-style: It would be cool to press `/`, then type some word or regex and hit enter to search for it. Bonus points if you can also do search next via `n` and `N` (or `*`). It's probably good enough to do case-insensitive search only.
- Exposing the HN Algolia Search in a separate tab would be cool. Not sure if there's an API for that.
- You might want to document that `ctrl+c` works as an exit key combo (on OSX at least). Pressing F12 is not all that intuitive to me and I think most other terminal apps that I use support either that and / or `q` (`less` is one of the annoying ones which suppresses `ctrl+c`).
- Change the name of the terminal tab from "Python" to "OSH" or smth like that. [This](https://stackoverflow.com/a/37690160/1174378) seems to work fine on OSX with [iTerm2](https://iterm2.com/) and the standard Terminal app.
- Maybe consider documenting the go-to-beginning / go-to-end keys. On OSX at least, I can use fn+up/down arrows to scroll one page at a time and fn+left/right arrows to scroll to beginning / end.

Small nitpicks:
- The "Updated x seconds ago" is kinda' distracting for the first 60 seconds. I'd consider updating it once / minute instead.
- Clicking in the header area where it says "Orange Site Hit" expands the header a bit, but I'm not sure I get why it does that.

**Comment by davep** [comment]
Working through these:

- Search, including local lists, comments in a comment view, and Algolia, are all already in the TODO list in the README. I'm very keen to add them when I get a moment and to make them as terminal-tool-normal as possible.
- Noted on the <kbd>Ctrl</kbd>+<kbd>C</kbd> thing; it's been second nature for me for decades in so many places I didn't even think to highlight it.
- I've never been a fan of apps that impose terminal title names (indeed, I have kitty set up such that it ignores any attempt to do so); but I can see why it might be handy for some. Pretty sure we actually do have an API for that in Textual somewhere so I'll see about that.
- macOS here myself, although I'm a [Keycron user](https://www.keychron.com/products/keychron-k8-tenkeyless-wireless-mechanical-keyboard) so just tend to bounce on the page and home/end keys, and that's the keys that the app uses. It's been a wee while since I used an Apple keyboard but I seem to remember such keys exist (or the equivalents).
- I'll consider making the update frequency for the "last updated" text configurable; although it's pretty common to have such human-friendly elapsed time work that way.
- Yeah, the header thing is a Textual weirdness that doesn't really do anything (I honestly have no clue *why* it's designed that way). I should probably subclass it and stop it doing that at least.

**Reply by davep** [reply]
Fun thing: of all the terminals I have to hand (kitty, iTerm, Terminal, WezTerm, Contour and rio) only one of them hints at Python in the window title.

![Screenshot 2024-01-17 at 20 57 46](https://github.com/davep/oshit/assets/28237/f54b4c99-4bbd-45bd-994f-b256752148f6)

Likely down to my shell/prompt config and/or terminal config, I guess.

**Comment by mihaitodor** [comment]
Oh, caught me! I didn't read the entire Readme 😅 Thank you for going through all of these!

> I've never been a fan of apps that impose terminal title names 

For me, it just shows "Python", which is fine, just thought it might be cooler to show smth else :)

> Fun thing: of all the terminals I have to hand (kitty, iTerm, Terminal, WezTerm, Contour and rio) only one of them hints at Python in the window title.

Wow, you have many! :) I'm using classic bash in iTerm2 and I'm sure I'm missing out on a lot of fun features... I will try switching to zsh next month when I'll get my shiny new laptop, but not sure if I'll bother to try other terminal apps as well.

**Reply by davep** [reply]
Heh! I'm a `fish` user myself (weirdly I was `zsh` all the way when I started using macOS, back when bash was the default, and then when I changed to `fish` macOS changed to `zsh` on the next release).

As for all the terminals... given I work on Textual I tend to try and test with as many as possible. ;-)
